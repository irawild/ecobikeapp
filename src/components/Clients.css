.clients {
    align-items: center;
    text-align: center;
}