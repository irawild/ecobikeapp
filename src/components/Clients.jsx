import './Clients.css'

const Clients = (props) => {
    if (props.clientsList)
    {
        console.log(props)
        return <div className="clients">
            <ul>
            {props.clientsList.map((client, i) => {
                return <il key={i}>{client.name}</il>
            })}
            </ul>
        </div>
    }
    else{
        return <div className="clients">No clients found</div>
    }
}

export default Clients