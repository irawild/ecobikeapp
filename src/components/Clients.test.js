import React from "react";
import renderer from "react-test-renderer";

import Clients from "./Clients";

describe("Snapshot testing section", () => {
  it("renders when there are no items", () => {
    const tree = renderer.create(<Clients />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders when there is one item", () => {
    const tree = renderer
      .create(<Clients clientsList={["One item"]} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders when there are multiple items", () => {
    const tree = renderer
      .create(
        <Clients clientsList={["Item 1", "Item 2", "Item 3", "Item 4"]} />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
