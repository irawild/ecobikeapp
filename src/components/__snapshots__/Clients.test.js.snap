// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot testing section renders when there are multiple items 1`] = `
<div
  className="clients"
>
  <ul>
    <il />
    <il />
    <il />
    <il />
  </ul>
</div>
`;

exports[`Snapshot testing section renders when there are no items 1`] = `
<div
  className="clients"
>
  No clients found
</div>
`;

exports[`Snapshot testing section renders when there is one item 1`] = `
<div
  className="clients"
>
  <ul>
    <il />
  </ul>
</div>
`;
